ClinGen/clincoded
# Connects to #871. Connects to #860. Connects to #854. Connects to #823.

This PR primarily consists of commits pertinent to change requests in both #860 and #871.

**Technical details:**
1. A new `clinvar_interpretations.js` file has been created to only include 2 helper methods used to fetch/parse ClinVar RCV data for the display of ClinVar Interpretations table on the Basic Information tab.
2. To fetch condition data for individual ClinVar RCV IDs, a second API call (to ClinVar) is required given the parsed RCV IDs from the initially parsed ClinVar variant data.
3. Karen had suggested an alternative API (e.g. http://myvariant.info/v1/query?q=RCV000049262&fields=clinvar.rcv) by myvariant.info because the response appears to have the same RCV condition data and it is already in JSON format. However, after another closer, we discovered that the response does not include all associated conditions for any given RCV ID. Rather, it only includes one condition. I suggested to Matt that he and Karen should get in touch with the myvariant.info team in this regard. Then we can switch to use myvariant.info from ClinVar for the second API call in the future when all conditions are included in the response.

**Steps to test:**
1. Login and use 55847 for the variant ID at the selection interface.
2. Proceed to the Basic Information tab.
3. Expect to see the previously named "ClinVar Transcripts" table now being renamed to "RefSeq Transcripts".
4. Expect to see a new "Exon" column added to the "ClinVar Primary Transcript", "RefSeq Transcripts", "Ensembl Transcripts" tables.
5. Expect to see the "RefSeq Transcripts" table being populated with RefSeq transcript data in the same format as seen in the "Ensembl Transcripts" table.
6. Expect to see the "ClinVar Primary Transcript" table also displaying Exon data.
7. Expect to see "* Canonical transcript" notes in the headers of "RefSeq Transcripts" and "Ensembl Transcripts" tables. And expect to see "*" indicators in the rows that are canonical transcripts in their respective tables.
8. Expect to see a new "ClinVar Interpretations" table added to the top of the Basic Information tab. And expect to see relevant RCV data being displayed in the table, as well as clicking on individual links in this table will open up new windows to the intended destinations.
